### ISSUE TITLE
Automatically parse the authors given a doi

### ISSUE THREAD

**Issue by Avsecz** [kipoi/website#77]
As suggested in https://github.com/greenelab/deep-review/issues/886#issuecomment-410030820, use https://github.com/greenelab/manubot to get the author list given a doi link.

## Tasks

- [x] check if the url is a doi link
- [x] get the author list for each model / model group
- [x] merge the authors with the original model entry
  - if first-last name match, keep the email / github information
  - if it doesn't match, add the information

**Comment by agitter** [comment]
@dhimmel is finishing some major Manubot refactoring that will make it easier to do this.  The API isn't stable yet, but it will be possible to do something like:
```
> from manubot.cite import citation_to_citeproc
> citation = citation_to_citeproc('doi:10.1101/375345')
> citation['title']
'Kipoi: accelerating the community exchange and reuse of predictive models for genomics'
> citation['author']
[{'given': 'Ziga', 'family': 'Avsec'}, {'given': 'Roman', 'family': 'Kreuzhuber'}, {'given': 'Johnny', 'family': 'Israeli'}, {'given': 'Nancy', 'family': 'Xu'}, {'given': 'Jun', 'family': 'Cheng'}, {'given': 'Avanti', 'family': 'Shrikumar'}, {'given': 'Abhimanyu', 'family': 'Banerjee'}, {'given': 'Daniel S', 'family': 'Kim'}, {'given': 'Lara', 'family': 'Urban'}, {'given': 'Anshul', 'family': 'Kundaje'}, {'given': 'Oliver', 'family': 'Stegle'}, {'given': 'Julien', 'family': 'Gagneur'}]
```

**Comment by Avsecz** [comment]
That's really nice! Can you ping me here again then it's so far? I'll go ahead and setup everything else.

**Comment by Avsecz** [comment]
Everything is ready. Let me know when it's ready. Two questions:
1. Do you have any good strategy on how to merge the existing authors listed in `model.yaml` with the parsed ones? I implemented a naive solution of overwriting the existing authors with the scraped ones. The problem could be where there are additional authors in `model.yaml` and where to put them in the order.
2. Does `citation_to_citeproc` work also with `https` links: `https://doi.org/10.1038/nmeth.3547`?

**Comment by agitter** [comment]
1. That's tricky, especially because some of your authors have annotations (GitHub usernames).  If you didn't have any annotations, I would suggest completely overwriting the authors in `model.yaml` and adding any authors not listed on the manuscript as contributors.  We don't provide any name matching in Manubot, so you would have to implement that yourself to do more complex merges with the yaml file authors and automatically extracted authors.  Note that authors sometimes are listed with separate `family` and `given` names and sometimes as a single `literal` name.  It depends on the metadata provider.
2. Table 2 of the [Manubot manuscript](https://greenelab.github.io/meta-review/#manubot) shows a little more of what `citation_to_citeproc` accepts.  The prefix `url:` or `doi:` is being used to retrieve information from a different metadata provider.  We could considering detecting URLs that could instead be resolved to a more specific source (DOI, arXiv, PubMed, and PubMed Central URLs), but that isn't implemented.  URLs are supported but use Greycite for metadata, which is much less reliable than DOI-based metadata.  Greycite gives errors with your example `https://doi.org/10.1038/nmeth.3547`.  Our typical strategy would be use a standard reference identifier in your `cite_as` field of `model.yaml`, such as `doi:10.1038/nmeth.3547` or `arxiv:1807.09946`.  Then you can pass the identifier directly to `citation_to_citeproc` and get the URL from the citation, e.g. `citation['URL']` in my example above.

If you want to quickly test extracting metadata from different manuscripts to see what you get back, you can also use Manubot from the command line:
```
manubot cite doi:10.1101/375345
```
or
```
manubot cite arxiv:1807.09946 --render
```
The `--render` option requires pandoc to generate the formatted reference.

I'm cross-referencing https://github.com/greenelab/manubot/pull/57 to remind me to follow up here.  Once that is merged, version 0.2.0 will be released with the API changes.

**Comment by Avsecz** [comment]
@agitter I managed to implement the feature. Looks really nice and I've actually spotted some a few spelling mistakes:) I'm using manubot from the most recent commit: https://github.com/kipoi/website/blob/master/app/requirements.txt#L5.

See https://kipoi.org/groups/. Thanks!

**Comment by agitter** [comment]
@Avsecz that's excellent that you've gotten it to work for this use case.  Even though it will work fine installing from GitHub, I'll still follow up once the next version is available on pip.

**Comment by dhimmel** [comment]
Nice I looked through https://github.com/kipoi/website/pull/78 to see the implementation. One potential Manubot feature that could be helpful would be "URL to citation" that could perform the following conversion:

https://doi.org/10.1101/gr.227819.117 ⟶ doi:10.1101/gr.227819.117
http://dx.doi.org/10.1101/gr.227819.117 ⟶ doi:10.1101/gr.227819.117
https://doi.org/gd54x8 ⟶ doi:10.1101/gr.227819.117
https://arxiv.org/pdf/1603.09123.pdf ⟶ arxiv:1603.09123

Would also hopefully work with other types of URLs. Not a priority for me at the moment, but something to keep in mind.

> I'll still follow up once the next version is available on pip.

https://github.com/greenelab/manubot/pull/56 may end up changing the import location for `citation_to_citeproc`... something to keep in mind. Won't be an issue unless you upgrade the manubot version in `requirements.txt`.

**Comment by Avsecz** [comment]
> One potential Manubot feature that could be helpful would be "URL to citation" that could perform the following conversion


@dhimmel that would be pretty handy! Let me know if/when it will get implemented or mention me in the issue proposing this feature so that I can subscribe to it.

**Comment by agitter** [comment]
> I'll still follow up once the next version is available on pip.

Manubot version 0.2.0 with these citations features is now on [PyPI](https://pypi.org/project/manubot/) in case you want to update the `requirements.txt` file.

**Comment by dhimmel** [comment]
Yep and the `citation_to_citeproc` import can stay the same:

https://github.com/kipoi/website/blob/278fffb53370295cc6ed848c8d14b107b2546962/app/models/authors.py#L43

**Comment by Avsecz** [comment]
Great. I updated it and it worked:) Thanks!:)
